eng/scripts/get-aspire-cli-pr.sh

```diff
@@ -728,6 +743,75 @@ download_aspire_cli() {
     return 0
 }
 
+# Function to check if VS Code CLI is available
+check_vscode_cli_dependency() {
+    if ! command -v code >/dev/null 2>&1; then
+        say_warn "VS Code CLI (code) is not available in PATH. Extension installation will be skipped."
+        say_info "To install VS Code extensions, ensure VS Code is installed and the 'code' command is available."
+        return 1
+    fi
+    return 0
+}
+
+# Function to download VS Code extension artifact
+download_aspire_extension() {
+    local workflow_run_id="$1"
+    local temp_dir="$2"
+    local download_dir="$temp_dir/extension"
+
+    say_info "Downloading VS Code extension from GitHub - $EXTENSION_ARTIFACT_NAME ..."
+
+    if [[ "$DRY_RUN" == true ]]; then
+        say_info "[DRY RUN] Would download extension artifact: $EXTENSION_ARTIFACT_NAME"
+        echo "$download_dir"
+        return 0
+    fi
+
+    mkdir -p "$download_dir"
+    if ! gh run download "$workflow_run_id" --name "$EXTENSION_ARTIFACT_NAME" --dir "$download_dir" --repo "$REPO"; then
+        say_warn "Failed to download VS Code extension artifact"
+        say_info "This could mean the extension artifact is not available for this build."
+        return 1
+    fi
+
+    echo "$download_dir"
+    return 0
+}
+
+# Function to install VS Code extension
+install_aspire_extension() {
```

Comment:
Some team members use `code-insiders` I think it would be good to allow people to target code insiders for specific builds.